gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependency.groovy

```diff
@@ -61,48 +57,38 @@ class ProductDependency implements Serializable {
             @Nullable String recommendedVersion) {
         this.productGroup = productGroup
         this.productName = productName
-        this.detectConstraints = false
         this.minimumVersion = minimumVersion
         this.maximumVersion = maximumVersion
         this.recommendedVersion = recommendedVersion
         isValid()
     }
 
     def isValid() {
-        if (detectConstraints) {
-            if (minimumVersion != null) {
-                throw new IllegalArgumentException("minimum version must not be specified if detectConstraints is enabled")
-            }
-            if (maximumVersion != null) {
-                throw new IllegalArgumentException("maximum version must not be specified if detectConstraints is enabled")
-            }
-            if (recommendedVersion != null) {
-                throw new IllegalArgumentException("recommended version must not be specified if detectConstraints is enabled")
-            }
-        } else {
-            if (minimumVersion == null) {
-                throw new IllegalArgumentException("minimum version must be specified");
-            }
-
-            [maximumVersion].each {
-                if (it && !SlsVersion.check(it) && !SlsVersionMatcher.safeValueOf(it).isPresent()) {
-                    throw new IllegalArgumentException(
-                            "maximumVersion must be valid SLS version or version matcher: " + it)
-                }
-            }
-
-            [minimumVersion, recommendedVersion].each {
-                if (it && !SlsVersion.check(it)) {
-                    throw new IllegalArgumentException(
-                            "minimumVersion and recommendedVersions must be valid SLS versions: " + it)
-                }
+        if (productGroup == null) {
+            throw new IllegalArgumentException("productGroup must be specified")
+        }
+        if (productName == null) {
+            throw new IllegalArgumentException("productName must be specified")
+        }
+        if (minimumVersion == null) {
+            throw new IllegalArgumentException("minimum version must be specified")
+        }
+        maximumVersion.with {
+            if (it == null || !SlsVersion.check(it) && !SlsVersionMatcher.safeValueOf(it).isPresent()) {
+                throw new IllegalArgumentException(
+                        "maximumVersion must be valid SLS version or version matcher: " + it)
             }
-
-            if (minimumVersion == maximumVersion) {
-                throw new IllegalArgumentException("minimumVersion and maximumVersion must be different "
-                        + "in product dependency on " + this.productName)
+        }
+        [minimumVersion, recommendedVersion].each {
```

Comment:
I think these short hands are undesirable, since they don't really improve readability of the code and make the error messages less precise